parser/src/fstring.rs

```diff
@@ -43,13 +43,49 @@ impl<'a> FStringParser<'a> {
                     })
                 }
                 ':' if delims.is_empty() => {
+                    let mut nested = false;
+                    let mut in_nested = false;
+                    let mut spec_expression = String::new();
                     while let Some(&next) = self.chars.peek() {
-                        if next != '}' {
-                            spec.push(next);
-                            self.chars.next();
-                        } else {
-                            break;
+                        match next {
+                            '{' => {
+                                if in_nested {
+                                    return Err(ExpressionNestedTooDeeply);
+                                } else {
```

Comment:
You don't need this `else`, as you return in the first condition.

Reply:
Done